src/server/api/controllers/admin/EventsController.ts

```diff
@@ -141,12 +147,74 @@ export class EventsController {
   @UseBefore(RoleAuth(Role.ROLE_ADMIN))
   @UseBefore(ValidateEventID)
   async getTeams(@AuthorisedAdmin() admin: Admin, @SelectedEventID() event: EventDocument): Promise<TESCTeam[]> {
+    const isOrganiser = await this.EventService.isAdminOrganiser(event.alias, admin);
+    if (!isOrganiser) {
+      throw new ForbiddenError(ErrorMessage.PERMISSION_ERROR());
+    }
+
+    const teams = await this.TeamService.getTeamsByEvent(event);
+    return Promise.all(teams.map(this.TeamService.populateTeammatesAdminFields))
+  }
+
+  @Patch('/:eventId/teams')
+  @UseBefore(RoleAuth(Role.ROLE_ADMIN))
+  @UseBefore(ValidateEventID)
+  async updateTeam(
+    @AuthorisedAdmin() admin: Admin,
+    @SelectedEventID() event: EventDocument,
+    @Body() req: UpdateTeamRequest
+  ) {
     const isOrganiser = await this.EventService.isAdminOrganiser(event.alias, admin);
     if (!isOrganiser) {
       throw new BadRequestError(ErrorMessage.PERMISSION_ERROR());
     }
 
-    return this.TeamService.getTeamsByEvent(event);
+    const teamId = req.team._id;
+    if (!teamId) throw new BadRequestError('The team ID must be included in the request');
+    
+    // Start a transaction since we are accessing many different documents
+    const transactionSession = await this.TransactionService.startTransaction();
+
+    let team: TeamDocument = await this.TeamService.getTeamById(teamId, transactionSession);
+    if (!team) throw new BadRequestError(`No team with ID ${teamId} found`);
+
+    // Add/Remove users based on patch body
+    const fetchUserByEmail = async email => {
+      const user = await this.UserService.getUserByEventAndEmail(email, event, false, transactionSession);
+      if (!user) throw new BadRequestError(ErrorMessage.NO_ACCOUNT_EXISTS());
```

Comment:
could we specify which emails aren't attached to an account in the error message?

Reply:
oops i missed panda's comment about this. this can be changed in the future :)